agent/background.go

```diff
@@ -50,58 +50,49 @@ func (a *Agent) startHeartbeat(ctx context.Context, tc *taskContext, heartbeat c
 	}
 }
 
-func (a *Agent) startIdleTimeoutWatch(ctx context.Context, tc *taskContext, timeout chan struct{}, resetIdleTimeout chan time.Duration) {
+func (a *Agent) startIdleTimeoutWatch(ctx context.Context, tc *taskContext, cancel context.CancelFunc) {
 	timeoutInterval := defaultIdleTimeout
```

Comment:
Do we need the defaultIdleTimeout at all? Should we delete it and use the defaultCmdTimeout instead?

Reply:
yay deleting things!